From 592581753cf7a281eafa251ddc976b0eede11afa Mon Sep 17 00:00:00 2001
From: Max Horn <max@quendi.de>
Date: Wed, 14 Sep 2022 00:12:02 +0200
Subject: [PATCH 4/4] kernel: workaround GCC 12 warning

With GCC 12, we get a spurious warning indicating that `buf` in `GetStr`
may be used uninitialized. Weirdly, it does not complain about the almost
identical code in e.g. `GetPragma`.
---
 src/scanner.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/scanner.c b/src/scanner.c
index f5dbbcc04..23adc944e 100644
--- a/src/scanner.c
+++ b/src/scanner.c
@@ -653,7 +653,7 @@ static Char GetEscapedChar(ScannerState * s)
 static Char GetStr(ScannerState * s, Char c)
 {
     Obj  string = 0;
-    Char buf[1024];
+    Char buf[1024] = "";
     UInt i = 0;
 
     while (c != '"' && c != '\n' && c != '\377') {
-- 
2.38.1

